locallibrary/catalog/views.py

```diff
@@ -54,3 +71,42 @@ class LoanedBooksByUserListView(LoginRequiredMixin,generic.ListView):
 
     def get_queryset(self):
         return BookInstance.objects.filter(borrower=self.request.user).filter(status__exact='o').order_by('due_back')
+
+@login_required
+@permission_required('catalog.can_mark_returned', raise_exception=True)
+def renew_book_librarian(request, pk):
+    """View function for renewing a specific BookInstance by librarian."""
+    book_instance = get_object_or_404(BookInstance, pk=pk)
+
+    if request.method == 'POST':
+        form = RenewBookForm(request.POST)
+
+        if form.is_valid():
+            book_instance.due_back = form.cleaned_data['renewal_date']
```

Comment:
`cleaned_data` thuộc tính này có tác dụng gì em?

Reply:
dạ để "làm sạch" dữ liệu được đưa vào vì có nhiều trường hợp là người dùng cố ý nhập dữ liệu không an toàn vào nhằm tấn công web ạ

Reply:
"dữ liệu sạch" là ntn hở em ??